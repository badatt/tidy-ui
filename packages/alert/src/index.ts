export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
