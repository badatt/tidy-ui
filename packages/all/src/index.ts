export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
