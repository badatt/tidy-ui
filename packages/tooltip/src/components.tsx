import { css, styled } from '@tidy-ui/commons';
import { ITooltipProps } from './types';

const bgColor = css`
  ${({ theme: { palette, isDark } }) => css`
    ${isDark ? palette.neutral[700] : palette.neutral[800]}
  `}
`;

const color = css`
  ${({ theme: { palette } }) => css`
    ${palette.neutral[100]}
  `}
`;

const caretStyle = css`
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
`;

const tooltipStyle = css`
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: fit-content;
  background-color: ${bgColor};
  color: ${color};
  z-index: 1000000;
  text-align: center;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
`;

//-------
const caretTop = css`
  ${caretStyle}
  top: calc(-5px);
  left: calc(50% - 5px);
  border-color: ${bgColor} transparent transparent transparent;
`;

const tooltipTop = css`
  ${tooltipStyle}
  top: calc(-0.8rem - 0.8em - 5px);
`;

//-------
const caretRight = css`
  ${caretStyle}
  top: calc(50% - 5px);
  left: calc(100% - 5px);
  border-color: transparent ${bgColor} transparent transparent;
`;

const tooltipRight = css`
  ${tooltipStyle}
  left: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
`;

//--------
const caretBottom = css`
  ${caretStyle}
  top: calc(100% - 5px);
  left: calc(50% - 5px);
  border-color: transparent transparent ${bgColor} transparent;
`;

const tooltipBottom = css`
  ${tooltipStyle}
  top: calc(100% + 5px);
`;

//-------
const caretLeft = css`
  ${caretStyle}
  top: calc(50% - 5px);
  left: calc(-5px);
  border-color: transparent transparent transparent ${bgColor};
`;

const tooltipLeft = css`
  ${tooltipStyle}
  right: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
`;

const TooltipRoot = styled.div<ITooltipProps>`
  position: relative;

  ${({ content, direction }) => css`
    &::before {
      display: none;
      content: '${content}';
      ${direction == 'top' && tooltipTop}
      ${direction == 'right' && tooltipRight}
      ${direction == 'bottom' && tooltipBottom}
      ${direction == 'left' && tooltipLeft}
    }
    &::after {
      display: none;
      ${direction == 'top' && caretTop}
      ${direction == 'right' && caretRight}
      ${direction == 'bottom' && caretBottom}
      ${direction == 'left' && caretLeft}
    }
    &:hover,
    &:active,
    &:focus {
      &::before,
      &::after {
        display: inline-block;
      }
    }
  `}
`;

export { TooltipRoot };
