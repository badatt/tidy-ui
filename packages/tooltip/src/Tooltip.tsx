import React from 'react';
import { TooltipRoot } from './components';
import { ITooltipProps } from './types';

const Tooltip = React.forwardRef<HTMLDivElement, ITooltipProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <TooltipRoot ref={ref} role="tooltip" {...rest}>
      {children}
    </TooltipRoot>
  );
});

Tooltip.defaultProps = {
  direction: 'bottom',
};

export { Tooltip };
