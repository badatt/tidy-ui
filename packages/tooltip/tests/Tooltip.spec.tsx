/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Tooltip } from '../src';

describe('Tooltip', () => {
  it('Basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="right">
          Right
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="top">
          Top
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="bottom">
          Bottom
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="left">
          Left
        </Tooltip>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="right">
          Right
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="top">
          Top
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="bottom">
          Bottom
        </Tooltip>
        <Tooltip content="Lorem ipsum dolor sit amet consectetur" direction="left">
          Left
        </Tooltip>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
