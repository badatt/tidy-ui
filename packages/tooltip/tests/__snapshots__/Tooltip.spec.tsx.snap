// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
}

.c0::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  left: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c0::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(100% - 5px);
  border-color: transparent   hsla(0,0%,15%,1)  transparent transparent;
}

.c0:hover::before,
.c0:active::before,
.c0:focus::before,
.c0:hover::after,
.c0:active::after,
.c0:focus::after {
  display: inline-block;
}

.c1 {
  position: relative;
}

.c1::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  bottom: calc(100% + 5px);
  left: 0;
}

.c1::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(-5px);
  left: calc(50% - 5px);
  border-color: hsla(0,0%,15%,1)  transparent transparent transparent;
}

.c1:hover::before,
.c1:active::before,
.c1:focus::before,
.c1:hover::after,
.c1:active::after,
.c1:focus::after {
  display: inline-block;
}

.c2 {
  position: relative;
}

.c2::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  top: calc(100% + 5px);
  left: 0;
}

.c2::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(100% - 5px);
  left: calc(50% - 5px);
  border-color: transparent transparent   hsla(0,0%,15%,1)  transparent;
}

.c2:hover::before,
.c2:active::before,
.c2:focus::before,
.c2:hover::after,
.c2:active::after,
.c2:focus::after {
  display: inline-block;
}

.c3 {
  position: relative;
}

.c3::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  right: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c3::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(-5px);
  border-color: transparent transparent transparent   hsla(0,0%,15%,1);
}

.c3:hover::before,
.c3:active::before,
.c3:focus::before,
.c3:hover::after,
.c3:active::after,
.c3:focus::after {
  display: inline-block;
}

<body>
    <div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c0"
        direction="right"
        role="tooltip"
      >
        Right
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c1"
        direction="top"
        role="tooltip"
      >
        Top
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c2"
        direction="bottom"
        role="tooltip"
      >
        Bottom
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c3"
        direction="left"
        role="tooltip"
      >
        Left
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
}

.c0::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  left: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c0::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(100% - 5px);
  border-color: transparent   hsla(0,0%,15%,1)  transparent transparent;
}

.c0:hover::before,
.c0:active::before,
.c0:focus::before,
.c0:hover::after,
.c0:active::after,
.c0:focus::after {
  display: inline-block;
}

.c1 {
  position: relative;
}

.c1::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  bottom: calc(100% + 5px);
  left: 0;
}

.c1::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(-5px);
  left: calc(50% - 5px);
  border-color: hsla(0,0%,15%,1)  transparent transparent transparent;
}

.c1:hover::before,
.c1:active::before,
.c1:focus::before,
.c1:hover::after,
.c1:active::after,
.c1:focus::after {
  display: inline-block;
}

.c2 {
  position: relative;
}

.c2::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  top: calc(100% + 5px);
  left: 0;
}

.c2::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(100% - 5px);
  left: calc(50% - 5px);
  border-color: transparent transparent   hsla(0,0%,15%,1)  transparent;
}

.c2:hover::before,
.c2:active::before,
.c2:focus::before,
.c2:hover::after,
.c2:active::after,
.c2:focus::after {
  display: inline-block;
}

.c3 {
  position: relative;
}

.c3::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,15%,1);
  color: hsla(0,0%,96%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  right: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c3::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(-5px);
  border-color: transparent transparent transparent   hsla(0,0%,15%,1);
}

.c3:hover::before,
.c3:active::before,
.c3:focus::before,
.c3:hover::after,
.c3:active::after,
.c3:focus::after {
  display: inline-block;
}

<div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c0"
      direction="right"
      role="tooltip"
    >
      Right
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c1"
      direction="top"
      role="tooltip"
    >
      Top
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c2"
      direction="bottom"
      role="tooltip"
    >
      Bottom
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c3"
      direction="left"
      role="tooltip"
    >
      Left
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
}

.c0::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  left: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c0::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(100% - 5px);
  border-color: transparent   hsla(0,0%,64%,1)  transparent transparent;
}

.c0:hover::before,
.c0:active::before,
.c0:focus::before,
.c0:hover::after,
.c0:active::after,
.c0:focus::after {
  display: inline-block;
}

.c1 {
  position: relative;
}

.c1::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  bottom: calc(100% + 5px);
  left: 0;
}

.c1::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(-5px);
  left: calc(50% - 5px);
  border-color: hsla(0,0%,64%,1)  transparent transparent transparent;
}

.c1:hover::before,
.c1:active::before,
.c1:focus::before,
.c1:hover::after,
.c1:active::after,
.c1:focus::after {
  display: inline-block;
}

.c2 {
  position: relative;
}

.c2::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  top: calc(100% + 5px);
  left: 0;
}

.c2::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(100% - 5px);
  left: calc(50% - 5px);
  border-color: transparent transparent   hsla(0,0%,64%,1)  transparent;
}

.c2:hover::before,
.c2:active::before,
.c2:focus::before,
.c2:hover::after,
.c2:active::after,
.c2:focus::after {
  display: inline-block;
}

.c3 {
  position: relative;
}

.c3::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  right: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c3::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(-5px);
  border-color: transparent transparent transparent   hsla(0,0%,64%,1);
}

.c3:hover::before,
.c3:active::before,
.c3:focus::before,
.c3:hover::after,
.c3:active::after,
.c3:focus::after {
  display: inline-block;
}

<body>
    <div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c0"
        direction="right"
        role="tooltip"
      >
        Right
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c1"
        direction="top"
        role="tooltip"
      >
        Top
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c2"
        direction="bottom"
        role="tooltip"
      >
        Bottom
      </div>
      <div
        aria-label="Lorem ipsum dolor sit amet consectetur"
        class="c3"
        direction="left"
        role="tooltip"
      >
        Left
      </div>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
}

.c0::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  left: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c0::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(100% - 5px);
  border-color: transparent   hsla(0,0%,64%,1)  transparent transparent;
}

.c0:hover::before,
.c0:active::before,
.c0:focus::before,
.c0:hover::after,
.c0:active::after,
.c0:focus::after {
  display: inline-block;
}

.c1 {
  position: relative;
}

.c1::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  bottom: calc(100% + 5px);
  left: 0;
}

.c1::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(-5px);
  left: calc(50% - 5px);
  border-color: hsla(0,0%,64%,1)  transparent transparent transparent;
}

.c1:hover::before,
.c1:active::before,
.c1:focus::before,
.c1:hover::after,
.c1:active::after,
.c1:focus::after {
  display: inline-block;
}

.c2 {
  position: relative;
}

.c2::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  top: calc(100% + 5px);
  left: 0;
}

.c2::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(100% - 5px);
  left: calc(50% - 5px);
  border-color: transparent transparent   hsla(0,0%,64%,1)  transparent;
}

.c2:hover::before,
.c2:active::before,
.c2:focus::before,
.c2:hover::after,
.c2:active::after,
.c2:focus::after {
  display: inline-block;
}

.c3 {
  position: relative;
}

.c3::before {
  display: none;
  content: attr(aria-label);
  position: absolute;
  font-size: 0.8rem;
  padding: 0.4em 0.6em;
  border-radius: 0.2rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: hsla(0,0%,64%,1);
  color: hsla(0,0%,9%,1);
  z-index: 1000000;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  overflow-wrap: break-word;
  white-space: pre;
  pointer-events: none;
  border-radius: 3px;
  right: calc(100% + 5px);
  top: calc(50% - 0.4rem - 0.4em);
}

.c3::after {
  display: none;
  content: '';
  position: absolute;
  z-index: 1000001;
  border-width: 5px;
  border-style: solid;
  width: 0px;
  height: 0px;
  pointer-events: none;
  top: calc(50% - 5px);
  left: calc(-5px);
  border-color: transparent transparent transparent   hsla(0,0%,64%,1);
}

.c3:hover::before,
.c3:active::before,
.c3:focus::before,
.c3:hover::after,
.c3:active::after,
.c3:focus::after {
  display: inline-block;
}

<div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c0"
      direction="right"
      role="tooltip"
    >
      Right
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c1"
      direction="top"
      role="tooltip"
    >
      Top
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c2"
      direction="bottom"
      role="tooltip"
    >
      Bottom
    </div>
    <div
      aria-label="Lorem ipsum dolor sit amet consectetur"
      class="c3"
      direction="left"
      role="tooltip"
    >
      Left
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
